### ISSUE TITLE
Operators are not properly checked

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/dev_compiler#18]
Adding the following test - see the "should be" cases below:

```
      class A {
        A operator *(B b) {}
        A operator /(B b) {}
        A operator ~/(B b) {}
        A operator %(B b) {}
        A operator +(B b) {}
        A operator -(B b) {}
        A operator <<(B b) {}
        A operator >>(B b) {}
        A operator &(B b) {}
        A operator ^(B b) {}
        A operator |(B b) {}
      }

      class B {
        A operator -(B b) {}
      }

      foo() => new A();

      test() {
        A a = new A();
        B b = new B();
        var c = foo();
        a = a * b;
        a = a * /*pass should be info:DownCast*/c;
        a = a / b;
        a = a ~/ b;
        a = a % b;
        a = a + b;
        a = a + /*pass should be severe:StaticTypeError*/a;
        a = a - b;
        b = /*severe:StaticTypeError*/b - b;
        a = a << b;
        a = a >> b;
        a = a & b;
        a = a ^ b;
        a = a | b;
        c = (/*pass should be warning:DynamicInvoke*/c + b);
      }
```
